### PR TITLE
Code and UT update for epoch time calculation within DateTimeConverter 

### LINKED ISSUE #381: epochToDateTime(): Incorrect Year Boundary Handling
The epochToDateTime() function has two issues affecting leap year boundary handling:

Bug 1: Pre-1970 Leap Year Jan 1 -> Returns 'nullptr'
- Example: Epoch -315,619,200 (1960-01-01) returns null instead of valid date
- Affected: 1960, 1956, 1952, 1948, etc. (all leap years before 1970 at Jan 1)
- Root Cause:During pre-1970 conversion, the year calculation is off by two years (1960 -> 1958), causing incorrect day normalization and a null result.

Bug 2: Post-1970 Leap Year Dec 31 Rolls over to Next Year
- Example:
    1. 2020-12-31 converts to 2021-01-01 (incorrect)
    2. 1972-12-31 converts to 1973-01-01 (incorrect)
- Affected: All leap years after 1970 at Dec 31 (~100+ dates)
- Root Cause: The leap-year adjustment and year rollover checks occur in the wrong sequence, causing a one-day overflow into the following year.

Affected Ranges
- Pre-1970 non-leap: Correct
- Post-1970 non-leap: Correct
- Pre-1970 leap mid-year (Jan 2-Dec 30): Correct : Boundary invalid
- Post-1970 leap Jan 1-Dec 30: Correct : Boundary invalid
- Post-1970 leap Dec 31: Affected
- Pre-1970 leap Jan 1: Affected

Test Reproduction:
To recreate issue : Test cases added in PR https://github.com/eclipse-score/baselibs/pull/135 EpochToDateTime_Pre1970_YearBoundary_DayOverflowRollover

### PR DESCRIPTION
**Pull request by Rutuja-Patil-Bosch**
Summary

This PR fixes inaccuracies in epoch time calculation within DateTimeConverter and enhances unit test coverage by adding relevant test cases.

Followup PR [#135 ](https://github.com/eclipse-score/baselibs/pull/135)

Changes

- Fixed off-by-one issue in date calculation
- Removed incorrect leap year adjustment logic
- Improved epoch conversion accuracy
- Updated and aligned unit test cases
- Performed minor code cleanup

Root Cause

The previous implementation applied incorrect leap year adjustments and had boundary condition issues, leading to inaccurate epoch calculations.

Testing
 
- Validated using updated unit tests
- Verified edge cases for date conversion and leap-year handling

Impact

- Improves correctness of DateTimeConverter calculations
- Strengthens validation through better test coverage
- No impact on external interfaces

Fixes #381

### CODE REVIEW

**Comment by gierer** [comment]
@Rutuja-Patil-Bosch 
- Please address the copilot review comments
- Please update the commit messages. They should explain what the commit changes. (instead of "Follow up PR for eclipse-score#135")
- Bot commits should be avoided as requested by @AlexanderLanin  https://github.com/eclipse-score/baselibs/pull/135#issuecomment-4204652107

**Comment by Rutuja-Patil-Bosch** [comment]
> @Rutuja-Patil-Bosch
> 
>     * Please address the copilot review comments
> 
>     * Please update the commit messages. They should explain what the commit changes. (instead of "Follow up PR for [Add round-trip and leap-year coverage tests for DateTimeConverter (pre/post 1970) #135](https://github.com/eclipse-score/baselibs/pull/135)")
> 
>     * Bot commits should be avoided as requested by @AlexanderLanin  [Add round-trip and leap-year coverage tests for DateTimeConverter (pre/post 1970) #135 (comment)](https://github.com/eclipse-score/baselibs/pull/135#issuecomment-4204652107)

@gierer 

1. I'll commit fix for copilot findings
2. I have updated commit message by Removing 'follow up PR....' details. Commit message describes details about code change in epoch time calculation & relevant test cases update for enhanced coverage. Follow up PR link has mentioned in the PR description to get track of earlier commits. Is this enough or should I update any point?
3. For comment from AlexanderLanin: discussed internally & ensured that no bot generated commits will be added.

File: score/datetime_converter/datetime_converter.cpp [hunk 1]
```diff
@@ -70,24 +85,7 @@ bool isValidDateTimeFormat(const std::shared_ptr<DateTimeType> dateTime)
...
-        }
-    }
-
-    if (dateTime->m_month == 4 || dateTime->m_month == 6 || dateTime->m_month == 9 || dateTime->m_month == 11)
-    {
-        return (dateTime->m_day <= 30);
-    }
-
-    return true;
+    return (dateTime->m_day <= daysInMonth(dateTime->m_year, dateTime->m_month));
```
[the remark this answers is not in the document]
Comment (Rutuja-Patil-Bosch) [reply score/datetime_converter/datetime_converter.cpp]: updated

File: score/datetime_converter/datetime_converter_test.cpp [hunk 2]
```diff
@@ -453,6 +453,99 @@ TEST_F(DateTimeConverterTest, test_yearIsLeap)
...
+}
+
+TEST_F(DateTimeConverterTest, EpochToDateTime_Pre1800_ReturnsNull)
+{
+    RecordProperty("Description",
+        "Verify that epoch values mapping outside supported year range return nullptr.");
+    RecordProperty("TestType", "requirements-based");
+
+    // This epoch maps to year 10000, which is above the supported upper bound (9999).
+    const time_t epochOutOfRange = static_cast<time_t>(253405000000);
```
[the remark this answers is not in the document]
Comment (Rutuja-Patil-Bosch) [reply score/datetime_converter/datetime_converter_test.cpp]: Updated test case name

File: score/datetime_converter/datetime_converter.cpp [hunk 3]
```diff
@@ -17,6 +17,21 @@ namespace score
 namespace common
 {
 
+std::int8_t daysInMonth(const std::int16_t year, const std::int8_t month)
```
[the remark this answers is not in the document]
Comment (Rutuja-Patil-Bosch) [reply score/datetime_converter/datetime_converter.cpp]: updated static for this api()

**Comment by Rutuja-Patil-Bosch** [comment]
@gierer : is anything else I should update for this PR?

**Comment by paulquiring** [comment]
Formatting checks fail. The Followup PR [https://github.com/eclipse-score/baselibs/pull/135](https://github.com/eclipse-score/baselibs/pull/url) mentioned in the description does not exist.

**Comment by Rutuja-Patil-Bosch** [comment]
> Formatting checks fail. The Followup PR [https://github.com/eclipse-score/baselibs/pull/135](https://github.com/eclipse-score/baselibs/pull/url) mentioned in the description does not exist.

@paulquiring : formatting checks are passing now, & link also updated properly

**Review by gierer — COMMENTED** [review]
This issue wasn't introduced by you, but would be still great if you could quickly fix it https://github.com/eclipse-score/baselibs/pull/230#discussion_r3586960521

**Comment by Rutuja-Patil-Bosch** [comment]
> This issue wasn't introduced by you, but would be still great if you could quickly fix it [#230 (comment)](https://github.com/eclipse-score/baselibs/pull/230#discussion_r3586960521)

@gierer: Sure, I'll fix

**Comment by anmittag** [comment]
@Rutuja-Patil-Bosch some checks are missing. Please check to adapt the PR to enable the next steps

**Comment by Rutuja-Patil-Bosch** [comment]
> @Rutuja-Patil-Bosch some checks are missing. Please check to adapt the PR to enable the next steps

Thanks for the feedback. I'm investigating the missing checks and will update the PR to ensure all required workflows are triggered and the PR can proceed.

**Comment by Rutuja-Patil-Bosch** [comment]
Please approve/run the pending QNX workflows so the required status checks can be reported?

**Comment by Rutuja-Patil-Bosch** [comment]
@gierer 
This PR was created from the 'main' branch of 'bgsw-contrib'. Since that branch is regularly synced with 'eclipse-score/main', commits from 'bgsw-contrib[bot]' are appearing in the PR history.

I'll avoid this in future contributions by creating PRs from a dedicated feature branch rather than from 'main'.

For this PR, how would you recommend proceeding? Would it be better to create a new PR from a clean feature branch, or is there another preferred way to remove the unwanted 'bgsw-contrib[bot]' commits and keep the history clean?

File: score/datetime_converter/datetime_converter.cpp [hunk 4]
```diff
@@ -70,24 +85,7 @@ bool isValidDateTimeFormat(const std::shared_ptr<DateTimeType> dateTime)
...
-        }
-    }
-
-    if (dateTime->m_month == 4 || dateTime->m_month == 6 || dateTime->m_month == 9 || dateTime->m_month == 11)
-    {
-        return (dateTime->m_day <= 30);
-    }
-
-    return true;
+    return (dateTime->m_day >= 1) && (dateTime->m_day <= daysInMonth(dateTime->m_year, dateTime->m_month));
```
Comment (gierer) [inline score/datetime_converter/datetime_converter.cpp]:
```suggestion
    return dateTime->m_day <= daysInMonth(dateTime->m_year, dateTime->m_month);
```

And change the check above in line 71 to 
else if (dateTime->m_day < 1 || dateTime->m_day > 31)

**Comment by gierer** [comment]
> @gierer This PR was created from the 'main' branch of 'bgsw-contrib'. Since that branch is regularly synced with 'eclipse-score/main', commits from 'bgsw-contrib[bot]' are appearing in the PR history.
> 
> I'll avoid this in future contributions by creating PRs from a dedicated feature branch rather than from 'main'.
> 
> For this PR, how would you recommend proceeding? Would it be better to create a new PR from a clean feature branch, or is there another preferred way to remove the unwanted 'bgsw-contrib[bot]' commits and keep the history clean?

@4og said that he will squash the commits once it's ready for merge

**Review by gierer — APPROVED** [review]

File: score/datetime_converter/datetime_converter_test.cpp [hunk 5]
```diff
@@ -453,6 +453,99 @@ TEST_F(DateTimeConverterTest, test_yearIsLeap)
...
+{
+    RecordProperty("Description", "Verify invalid day values are rejected.");
+    RecordProperty("TestType", "requirements-based");
+
+    auto dtt = std::make_shared<DateTimeType>(2020, 12, 34, 3, 4, 5);
+    ASSERT_FALSE(score::common::isValidDateTimeFormat(dtt));
+
+    dtt->m_day = -1;
+    ASSERT_FALSE(score::common::isValidDateTimeFormat(dtt));
+}
```
[the remark this answers is not in the document]
Comment (Rutuja-Patil-Bosch) [reply score/datetime_converter/datetime_converter_test.cpp:484]: Updated

**Review by 4og — APPROVED** [review]

**Comment by Rutuja-Patil-Bosch** [comment]
@4og I can see changes present in this PR are merged into main with PR https://github.com/eclipse-score/baselibs/pull/418. Should I close this PR?

**Comment by 4og** [comment]
> @4og I can see changes present in this PR are merged into main with PR #418. Should I close this PR?

yes
